x-pack/plugins/apm/public/components/shared/KueryBar/view.js

```diff
@@ -59,7 +59,9 @@ class KueryBarView extends Component {
         selectionStart,
         indexPattern,
         boolFilter
-      )).filter(suggestion => !startsWith(suggestion.text, 'span.'));
+      ))
+        .filter(suggestion => !startsWith(suggestion.text, 'span.'))
+        .slice(0, 15);
```

Comment:
I noticed that rendering of the suggestions was super slow because the list can get very long. Limiting the number of suggestions to 15 reduced rendering time from 300ms to 20ms.